### DISCUSSION TITLE
Configuration changes

### DISCUSSION THREAD

**Discussion by jcqli** [DanielMartinus/Konfetti#234] [Q&A]
Does Konfetti currently support configuration changes? We want to be able to continue the animation through a configuration change (e.g. screen rotation) with largely the same visual and experience as before the configuration change (we want it to continue animating in the same way as before the change). Is this possible?

**Comment by DanielMartinus** [comment]
@jcqli that's a good question. Since all the calculations currently happen in `KonfettiView` and the UI is recreated when a configuration change happens it will mean Konfetti doesn't support configuration changes so all the drawing operations will be cleared from the screen. However, I think it's good to take this into account and I should start looking at a fix for this.

**Reply by DanielMartinus** [reply]
For example, I'm planning to separate the core of Konfetti where all calculations happen from the UI. This is one step into the direction of supporting configuration changes and restoring the state.
